.github/workflows/nx-cloud-agents.yml

```diff
@@ -128,20 +129,34 @@ jobs:
           restore-keys: |
             ${{ runner.os }}-node-${{ steps.versions.outputs.node_version }}-
 
-      - name: Get yarn cache directory path
-        if: steps.package_manager.outputs.name == 'yarn'
+      - name: Get yarn cache directory path [yarn 1.x]
+        if: steps.package_manager.outputs.name == 'yarn' && startsWith(steps.versions.outputs.yarn_version, '1.')
         id: yarn-cache-dir-path
         run: echo "dir=$(yarn cache dir)" >> $GITHUB_OUTPUT
 
-      - name: Use the node_modules cache if available [yarn]
-        if: steps.package_manager.outputs.name == 'yarn'
+      - name: Use the node_modules cache if available [yarn 1.x]
+        if: steps.package_manager.outputs.name == 'yarn' && startsWith(steps.versions.outputs.yarn_version, '1.')
         uses: actions/cache@v3
         with:
           path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
           key: ${{ runner.os }}-node-${{ steps.versions.outputs.node_version }}-yarn-${{ hashFiles('**/yarn.lock') }}
           restore-keys: |
             ${{ runner.os }}-node-${{ steps.versions.outputs.node_version }}-yarn-
 
+      - name: Get yarn cache directory path [yarn berry]
+        if: steps.package_manager.outputs.name == 'yarn' && !startsWith(steps.versions.outputs.yarn_version, '1.')
+        id: yarn-berry-cache-dir-path
+        run: echo "dir=$(yarn config get cacheFolder)" >> $GITHUB_OUTPUT
+
+      - name: Use the node_modules cache if available [yarn berry]
+        if: steps.package_manager.outputs.name == 'yarn' && !startsWith(steps.versions.outputs.yarn_version, '1.')
+        uses: actions/cache@v3
+        with:
+          path: ${{ steps.yarn-berry-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
```

Comment:
please include the node version in the cache key like we do in all other cases

Reply:
AFIK this is not needed for yarn berry: https://github.com/actions/setup-node/issues/325#issuecomment-916902678

Reply:
It's not _needed_ for any of them, but it is much better :)

Reply:
You don't change the node version of your project very often, but when you do it is most definitely worth busting caches over